main.cpp

```diff
@@ -1,25 +1,57 @@
 #include <iostream>
+#include <getopt.h>
 #include "common.h"
-#include "md5-duplicate-detection.h"
 #include "encrypt-decrypt.h"
 #include "files.h"
 
-int main() {
-    std::cout << "Hello, World!" << std::endl;
-    /*
-    std::string directory = "some directory";
+int main(int argc, char *argv[]) {
+
+    int opt;
+    const char *directory;
     std::vector<fs::path> files;
-    getFiles(directory, files);
-    for (fs::path i: files)
-        std::cout << i.string() << ' ';
-        */
 
-//    fs::path A("test-files/A");
-//    fs::path B("test-files/B");
-//    is_data_duplicate(A,B);
+    while (true) { //getopt creates a data-structure that can be looped through from the argv array to process the parameters and their arguments
+
+        int option_index = 0;
+        static struct option long_options[] =
+                {
+                        {"help",     no_argument, NULL,  'h' },
+                        {"encrypt",  no_argument, NULL,  'e' },
+                        {"decrypt",  no_argument, NULL,  'd' },
+                        {NULL,      0,    NULL,   0  }
+                };
+        opt = getopt_long(argc, argv, "he:d:", long_options, &option_index);
+
+        if(opt == -1)
+            break;
+
+        switch (opt) {
+            case 'h':
+                printf("ex1 -e --encrypt -d --decrypt\n"
+                       "-e --encrypt \".../path-to-root\"  | Encrypts files in a given directory and accepts the root path as an argument\n"
+                       "-d --decrypt \".../path-to-root\"  | Decrypts encrypted files from this program and accepts the root path as an argument\n");    //placeholder
```

Comment:
Commands generally have a standard format as you can see in this example
```
Usage: ls [OPTION]... [FILE]...
List information about the FILEs (the current directory by default).
Sort entries alphabetically if none of -cftuvSUX nor --sort is specified.

Mandatory arguments to long options are mandatory for short options too.
  -a, --all                  do not ignore entries starting with .
  -A, --almost-all           do not list implied . and ..
      --author               with -l, print the author of each file
  -b, --escape               print C-style escapes for nongraphic characters
      --block-size=SIZE      with -l, scale sizes by SIZE when printing them;
                               e.g., '--block-size=M'; see SIZE format below

```
I think all you're really missing is a proper usage line, and some commas.